### PR TITLE
System exit cleanup

### PR DESCRIPTION
**Pull request by ewels**
WIP PR as I try to address #78.

I'm basically doing three things as I go through each command:

* Move as much code out of `__main__.py` as possible into the module files, under a new function call
* Raise new custom exception types in case of errors instead of calling system exits
* Catch those exception types in `__main__.py` and log the messages, call a system exit.

The logic is that basically no other tool will ever import `__main__.py` so it's fine to put command-line specific code there such as `click` and hard exit calls. All of the main functionality goes into modules that can be imported and reused by other code bases, using exceptions that can be caught and dealt with differently elsewhere if needed.

While I'm going through, I will try to convert rich console print statements to log calls where I see them.

### CODE REVIEW

**Comment by ewels** [comment]
So that this refactoring doesn't get too huge, I will continue the other parts in separate branches and wait for this to be merged before going any further. Then I will do one PR per section.

So feel free to review + test 😃

**Comment by ewels** [comment]
Ok I lied, I did `dds ls` too. Bit it was a small one. Bed now.. 😅

**Comment by ewels** [comment]
I noticed that the package version was defined in two locations - fixed in 0f9af86 so that there is just one: in `setup.py`.

**Comment by ewels** [comment]
Ok, _really_ done with this PR now.. 😅 

Will branch before I touch anything else (barring PR review comments).

File: dds_cli/__main__.py [hunk 1]
```diff
@@ -201,117 +195,22 @@ def put(
...
+            username,
+            project,
+            source,
+            source_path_file,
+            break_on_fail,
+            overwrite,
+            num_threads,
+            silent,
+        )
+    except (dds_cli.exceptions.AuthenticationError, dds_cli.exceptions.UploadError) as e:
```
Comment (inaod568) [inline dds_cli/__main__.py:211]: I get the following error: 
```
File "/Users/inaod568/repos/DDS/dds_cli/dds_cli/exceptions.py", line 11, in __str__
    return f"\n:no_entry_sign: {self.message} :no_entry_sign:\n"
AttributeError: 'AuthenticationError' object has no attribute 'message'
```

Don't have time to fix it now though, sorry

Comment (ewels) [reply dds_cli/__main__.py:211]: How did you manage to trigger this? The place it is raised is in the `if self.method != "put":` check and I don't know how to make that fail.. 😅  (so struggling to replicate to fix)

**Review by inaod568 — APPROVED** [review]

**Pull request merged**